shared/actions/gregor.js

```diff
@@ -63,31 +63,33 @@ const setupEngineListeners = () => {
 
   // Filter this firehose down to the two systems we care about: "git", and "kbfs.favorites"
   // If ever you want to get OOBMs for a different system, then you need to enter it here.
-  RPCTypes.delegateUiCtlRegisterGregorFirehoseFilteredRpcPromise({systems: ['git', 'kbfs.favorites']})
-    .then(response => {
-      logger.info('Registered gregor listener')
-    })
-    .catch(error => {
-      logger.warn('error in registering gregor listener: ', error)
-    })
+  engine().actionOnConnect('registerGregorFirehose', () =>
+    RPCTypes.delegateUiCtlRegisterGregorFirehoseFilteredRpcPromise({systems: ['git', 'kbfs.favorites']})
+      .then(response => {
+        logger.info('Registered gregor listener')
+      })
+      .catch(error => {
+        logger.warn('error in registering gregor listener: ', error)
+      })
+  )
 
-  // The startReachibility RPC call both starts and returns the current
+  // The startReachability RPC call both starts and returns the current
   // reachability state. Then we'll get updates of changes from this state
   // via reachabilityChanged.
   // This should be run on app start and service re-connect in case the
   // service somehow crashed or was restarted manually.
-  return Saga.call(function*() {
-    const action = yield RPCTypes.reachabilityStartReachabilityRpcPromise()
+  engine().actionOnConnect('startReachability', () => {
+    const action = RPCTypes.reachabilityStartReachabilityRpcPromise()
```

Comment:
looking at this again, I'm not sure this is correct after all...let's chat with @chrisnojima tomorrow about this. The problem is we want to not only resolve the promise at some point in the future, but then dispatch an action when it's done. Saga.call might not be typed enough to be able to just return that value.